File: device-app/ios-swift/umsDemo/ums/Data/DataSource/UserLocalDataSource.swift
class UserLocalDataSource: JsonLocalDataSource<DataDef.User> {
    private static let fileName = "user.json"

    static let shared = UserLocalDataSource()

    private init() {
        super.init(Self.fileName)
    }

    var user: DataDef.User? {
        get {
            return value
        }
        set {
            value = newValue!
        }
    }
}

File: device-app/android/Java/umsDemo/app/src/main/java/com/rationalowl/umsdemo/data/datasource/UserLocalDataSource.java
package com.rationalowl.umsdemo.data.datasource;

import com.rationalowl.umsdemo.data.DataDef;

public class UserLocalDataSource extends JsonLocalDataSource<DataDef.User> {
    private static final String FILE_NAME = "user.json";

    private static UserLocalDataSource instance;

    public static synchronized UserLocalDataSource getInstance() {
        if (instance == null) {
            instance = new UserLocalDataSource();
        }

        return instance;
    }

    private UserLocalDataSource() {
        super(DataDef.User.class, FILE_NAME);
    }

    public DataDef.User getUser() {
        return getValue();
    }

    public void setUser(DataDef.User user) {
        setValue(user);
    }
}
